src/Maker/MakeForgottenPassword.php

```diff
@@ -0,0 +1,320 @@
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Renderer\FormTypeRenderer;
+use Symfony\Component\Validator\Validation;
+use Symfony\Bundle\SwiftmailerBundle\SwiftmailerBundle;
+
+/**
+ * @author Romaric Drigon <romaric.drigon@gmail.com>
+ *
+ * @internal
+ */
+final class MakeForgottenPassword extends AbstractMaker
+{
+    private $fileManager;
+
+    private $formTypeRenderer;
+
+    private $router;
+
+    public function __construct(FileManager $fileManager, FormTypeRenderer $formTypeRenderer, RouterInterface $router)
+    {
+        $this->fileManager = $fileManager;
+        $this->formTypeRenderer = $formTypeRenderer;
+        $this->router = $router;
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:forgotten-password';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig)
+    {
+        $command
+            ->setDescription('Creates a "forgotten password" mechanism')
+            //->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeRegistrationForm.txt'))
+        ;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        // initialize arguments & commands that are internal (i.e. meant only to be asked)
+        $command
+            ->addArgument('user-class')
+            ->addArgument('username-field')
+            ->addArgument('password-field')
+        ;
+
+        $interactiveSecurityHelper = new InteractiveSecurityHelper();
+
+        if (!$this->fileManager->fileExists($path = 'config/packages/security.yaml')) {
+            throw new RuntimeCommandException('The file "config/packages/security.yaml" does not exist. This command needs that file to accurately build the forgotten password form.');
+        }
+
+        $manipulator = new YamlSourceManipulator($this->fileManager->getFileContents($path));
+        $securityData = $manipulator->getData();
+        $providersData = $securityData['security']['providers'] ?? [];
+
+        $input->setArgument(
+            'user-class',
+            $userClass = $interactiveSecurityHelper->guessUserClass(
+                $io,
+                $providersData,
+                'Enter the User class which should be able to use "forgotten password" feature (e.g. <fg=yellow>App\\Entity\\User</>)'
+            )
+        );
+        $io->text(sprintf('Implementing forgotten password for <info>%s</info>', $userClass));
+
+        $input->setArgument(
+            'username-field',
+            $interactiveSecurityHelper->guessUserNameField($io, $userClass, $providersData)
+        );
+        $input->setArgument(
+            'password-field',
+            $interactiveSecurityHelper->guessPasswordField($io, $userClass)
+        );
```

Comment:
Because we're only using this to guess the setter, we could instead just guess the setter name and, if it's not found, ask the user to literally tell us the method name. For example, maybe we look for a `setPassword()` method. If that's found, we just use it. If it's not found, we ask "Enter the setter method on your User class that's used to set the encoded password (e.g. setPassword())".

Reply:
Agreed, I did that, and for e-mail getter too. I did not add `guessPasswordSetter` and `guessEmailGetter` to `InteractiveSecurityHelper`, because their re-usability would be limited.